### PR TITLE
ADM-442 Report page Classification section calculation

### PR DESCRIPTION
**Pull request by spadio9587**
## Summary
Classification calculation for report page
Classification

...

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

File: backend/src/main/java/heartbeat/service/report/CalculateClassification.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+				countMap.put(NONE_KEY, countMap.get(NONE_KEY) - 1);
+			}
+		}
+	}
+
+	private static String pickDisplayNameFromObj(Object object) {
+		if (object instanceof ICardFieldDisplayName) {
+			return ((ICardFieldDisplayName) object).getDisplayName();
+		}
+		// else if (object instanceof List) {
```
Comment (fengxinhou) [inline backend/src/main/java/heartbeat/service/report/CalculateClassification.java]: delete useless code

File: backend/src/main/java/heartbeat/service/report/CalculateClassification.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
```
Comment (fengxinhou) [inline backend/src/main/java/heartbeat/service/report/CalculateClassification.java]: delete useless import

File: backend/src/main/java/heartbeat/service/report/GenerateReporterService.java [hunk 3]
```diff
@@ -26,6 +26,8 @@ public class GenerateReporterService {
 
 	private final JiraService jiraService;
 
+	private final CalculateClassification calculateClassification;
```
Comment (fengxinhou) [inline backend/src/main/java/heartbeat/service/report/GenerateReporterService.java]: Class name cannot start with a verb？

File: backend/src/test/java/heartbeat/service/report/CalculateClassificationTest.java [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+import org.mockito.quality.Strictness;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class CalculateClassificationTest {
```
Comment (fengxinhou) [inline backend/src/test/java/heartbeat/service/report/CalculateClassificationTest.java]: rename CalculateClassificationTest

File: backend/src/test/java/heartbeat/service/report/GenerateReporterServiceTest.java [hunk 5]
```diff
@@ -78,10 +87,57 @@ void shouldReturnGenerateReportResponseWhenCallGenerateReporter() {
 
 		ReportResponse result = generateReporterService.generateReporter(request);
 		Velocity velocity = Velocity.builder().velocityForSP("0").velocityForCards("0").build();
-
 		assertThat(result).isEqualTo(ReportResponse.builder().velocity(velocity).build());
 	}
 
+	@Test
+
```
Comment (fengxinhou) [inline backend/src/test/java/heartbeat/service/report/GenerateReporterServiceTest.java:94]: delete blank line

File: backend/src/main/java/heartbeat/controller/report/GenerateReportController.java
Comment (TingLiMio) [inline backend/src/main/java/heartbeat/controller/report/GenerateReportController.java:1]: import * is not standardized

File: backend/src/main/java/heartbeat/controller/report/dto/response/Classification.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class Classification {
+
+	private String fieldName;
+
+	private List<ClassificationNameValuePair> pairs;
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/controller/report/dto/response/Classification.java:18]: rename `pairs` to `pairList`

File: backend/src/main/java/heartbeat/controller/report/GenerateReportController.java [hunk 7]
```diff
@@ -6,10 +6,7 @@
 import lombok.RequiredArgsConstructor;
 import lombok.extern.log4j.Log4j2;
 import org.springframework.validation.annotation.Validated;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/controller/report/GenerateReportController.java:9]: Should not use `.*`

File: backend/src/main/java/heartbeat/controller/report/dto/response/ClassificationNameValuePair.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class ClassificationNameValuePair {
+
+	private String name;
+
+	private String value;
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/controller/report/dto/response/ClassificationNameValuePair.java:16]: How to convert `50%` into a number, `0.5` is better than a string

**Review by guzhongren — APPROVED** [review]
Change with the comments
align the `string to number` issue with your TL

**Review by fengxinhou — APPROVED** [review]
LGTM

**Pull request merged**